platform/gce.go

```diff
@@ -44,10 +46,11 @@ type GCEOptions struct {
 }
 
 type gceCluster struct {
-	api      *compute.Service
-	sshAgent *network.SSHAgent
-	conf     *GCEOptions
-	machines map[string]*gceMachine
+	api        *compute.Service
+	sshAgent   *network.SSHAgent
+	conf       *GCEOptions
+	machines   map[string]*gceMachine
+	sync.Mutex // protects only against NewMachine being called in parallel
```

Comment:
Generally it's better to associate locks with the data they protect, not code.
At a glance, it looks like the mutex serializes access to the machines map.

I presume you left Destroy() alone because it needs to own the gceCluster instance to destroy it safely, but it's kind of an exception I'd avoid and throw the lock in anyway for consistency sake.

Machines() isn't being serialized, what's the rationale behind that?